nss-day-cohort-41/nutshell-a-red-red-rose-by-robert-burns
# Th login fetch

Description
Added login and registration functionality. At the moment it is disabled due to us not immediately pulling up login on start up but if you want to test it. You can add renderLogin() to the main instead of the Dashboard renders. Upon the login rendering the user will be prompted to login, if they need to register they can click the new user registration button. If they have an account they can type in their email or username and password to login. If that user doesn't exist or the password is incorrect the user will be notified and stay on the login page upon clicking the login or hitting enter. 

Upon going to the registration page, the user will be prompted to enter an email, username, password, and confirm their password. If the email doesn't contain "@" the user is prompted to enter a valid email, if the username is already in use they are prompted to use another, if the passwords do not match they are prompted to retype them. If all fields are correct then when the register button is clicked or the user hits enter the active user will be set to the new user, their user object will be pushed to the user API and the main Dashboard will be rendered to the Dom.

Fixes # (issue)
## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
# Testing Instructions
Start up the userInfo.APi
On the login page typing in incorrect info should be presented with window alerts regarding the issue. If the user info is correct username: "TestUser" password: "123" the user should be able to login and render the dashboard to the Dom.

The Register new user button should render the registration page. Typing incorrect info should cause window alerts regarding the issue. If the username is unique, the email conatains "@", and the passwords match, the register button should push the user object to the database, set the active user, and render the Dashboard to the Dom.

# Checklist:
- [ x] My code follows the style guidelines of this project
- [ x] I have performed a self-review of my own code
- [x ] I have commented my code, particularly in hard-to-understand areas
- [x ] My changes generate no new warnings or errors
- [x ] I have added test instructions that prove my fix is effective or that my feature works
